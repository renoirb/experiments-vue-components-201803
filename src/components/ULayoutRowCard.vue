<template>
  <v-layout
    row
    wrap
  >
    <v-flex>
      <v-card>
        <v-card-title
          v-if="showTitle"
          primary-title
        >
          <h2>{{ title }}</h2>
          <v-spacer />
          <slot name="top" />
        </v-card-title>
        <section
          class="card__main"
        >
          <slot />
        </section>
      </v-card>
    </v-flex>
  </v-layout>
</template>

<script>
import {
  VLayout,
  VFlex,
  VSpacer,
} from 'vuetify/es5/components/VGrid'
import VCard from 'vuetify/es5/components/VCard'

// @vue/component
export default {
  name: 'u-layout-row-card',
  props: {
    title: {
      type: [
        String,
        null,
      ],
      default: null,
    },
  },
  computed: {
    showTitle () {
      return typeof this.title === 'string'
    },
  },
  components: {
    VLayout,
    VFlex,
    VSpacer,
    VCard,
  },
}
</script>
