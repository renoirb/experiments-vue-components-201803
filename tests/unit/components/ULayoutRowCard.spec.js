import {
  mount,
} from '@vue/test-utils'

import ULayoutRowCard from '@/components/ULayoutRowCard.vue'

describe(`${ULayoutRowCard.name}`, () => {
  it('Parses markdown', () => {
    const title = `How much wood would a woodchuck chuck if a woodchuck could chuck wood?`
    const wrapper = mount(ULayoutRowCard, {
      propsData: {
        title,
      },
    })
    expect(wrapper.element).toMatchSnapshot()
  })
})
