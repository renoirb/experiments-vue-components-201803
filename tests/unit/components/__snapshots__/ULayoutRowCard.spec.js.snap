// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`u-layout-row-card Parses markdown 1`] = `
<div
  class="layout row wrap"
>
  <div
    class="flex"
  >
    <div
      class="card"
      style="height: auto;"
    >
      <v-card-title
        primary-title=""
      >
        <h2>
          How much wood would a woodchuck chuck if a woodchuck could chuck wood?
        </h2>
         
        <div
          class="spacer"
        />
         
      </v-card-title>
       
      <section
        class="card__main"
      />
    </div>
  </div>
</div>
`;
